core/arch/arm/plat-imx/main.c

```diff
@@ -86,10 +86,8 @@ void main_init_gic(void)
 	vaddr_t gicc_base;
 	vaddr_t gicd_base;
 
-	gicc_base = (vaddr_t)phys_to_virt(GIC_BASE + GICC_OFFSET,
-					  MEM_AREA_IO_SEC);
-	gicd_base = (vaddr_t)phys_to_virt(GIC_BASE + GICD_OFFSET,
-					  MEM_AREA_IO_SEC);
+	gicc_base = core_mmu_get_io_va(GIC_BASE + GICC_OFFSET);
+	gicd_base = core_mmu_get_io_va(GIC_BASE + GICD_OFFSET);
```

Comment:
i feel we could insure gic is secure.
Some assert should check gic was mapped secure. may at gic inits.